openchemistry/api.py

```diff
@@ -194,15 +194,21 @@ def find_spectra(identifier, stype='IR', source='NIST'):
     """
     source = source.lower()
 
+    inchi = GirderClient().get('molecules/{}'.format(identifier))["inchi"]
+
     params = {
-            'inchi' : identifier,
+            'inchi' : inchi,
             'spectrum_type' : stype,
             'source' : source
     }
 
-    spectra = GirderClient().get('experiments', parameters=params)
-
-    return spectra
+    spectrum = GirderClient().get('experiments', parameters=params)
```

Comment:
Will a spectrum always be found? If not we should guard against it being `None`.

Reply:
No, a spectrum will not always be found. I read the `.get` documentation and this will not raise a `KeyError` and will instead return `None` when no spectrum is returned. I would say this is the expected behavior. Do you agree?

Reply:
That is fine, but below you are trying to lookup keys in `spectrum` and it its `None` you will get an error ...

Reply:
You are right. Let me fix that. I had not understood completely.

Reply:
This is done.